h2/src/main/org/h2/value/ValueArray.java

```diff
@@ -107,7 +130,7 @@ public int compareTypeSafe(Value o, CompareMode mode, CastDataProvider provider)
     @Override
     public Object getObject() {
         int len = values.length;
-        Object[] list = (Object[]) Array.newInstance(componentType, len);
+        Object[] list = new Object[len];
```

Comment:
if this is used to return a value from a resultset, then it's an API change too

Reply:
H2 didn't preserve the class name, it was here only while object was in memory. And we discussed that. Our documentation also talks about the `Object[]`.

Reply:
I created issue #2190 to continue this discussion.